smartfs.lua

```diff
@@ -279,34 +380,89 @@ local function _sfs_recieve_(state,name,fields)
 	end
 	for key,val in pairs(state._ele) do
 		if val.submit then
-			if (val:submit(fields)==true) then
-				return true
-			end
+			val:submit(fields, player)
 		end
 	end
-	if state.closed ~= true then
-		state:_show_()
+
+	-- call onInput hook if enabled
+	if state._onInput then
+		state:_onInput(fields, player)
+	end
+
+	if not state.closed then
+		if state.location.type == "nodemeta" then
+			state:_attach_nodemeta_()
+		else
+			state:_show_()
+		end
 	else
-		minetest.show_formspec(name,"","size[5,1]label[0,0;Formspec closing not yet created!]")
-		if not state.is_inv then
-			smartfs.opened[name] = nil
+--		really? I was able to close window trough button_exit[]. Disabled for testing
+--		minetest.show_formspec(name,"","size[5,1]label[0,0;Formspec closing not yet created!]")
+		state.players:disconnect(player)
+		if state.location.type == "player" then
+			smartfs.opened[player:get_player_name()] = nil
 		end
+		return true
 	end
 	return true
 end
 
+-- Receive input from sender to the node form
+function smartfs.nodemeta_on_receive_fields(nodepos, formname, fields, sender)
+
+	-- get form info and check if it's a smartfs one
+	local meta = minetest.env:get_meta(nodepos)
+	local nodeform = meta:get_string("smartfs_name")
+	if not nodeform then -- execute only if it is smartfs form
+		print("SmartFS - (Warning) smartfs.nodemeta_on_receive_fields for node without smarfs data")
+		return false
+	end
+
+	-- get the currentsmartfs state
+	local opened_id = minetest.pos_to_string(nodepos)
+	local state = nil
+	local form = smartfs:__call(nodeform)
+	if not smartfs.opened[opened_id] or      --if opened first time
+	       smartfs.opened[opened_id].def.name ~= nodeform then --or form is changed
```

Comment:
A question about this indent. There should be a difference if the next line is the first command in the loop or if it is a part of the long condition. Or is there no visible difference in lua coding style?